torch/csrc/jit/tensorexpr/block_codegen.cpp

```diff
@@ -0,0 +1,319 @@
+#include <torch/csrc/jit/tensorexpr/block_codegen.h>
+
+#include <torch/csrc/jit/tensorexpr/analysis.h>
+#include <torch/csrc/jit/tensorexpr/eval.h>
+#include <torch/csrc/jit/tensorexpr/exceptions.h>
+#include <torch/csrc/jit/tensorexpr/execution_counter.h>
+#include <torch/csrc/jit/tensorexpr/ir_simplifier.h>
+
+#define DEBUG_PRINT 1
```

Comment:
Minor: it is generally a bad practice to send out code with DEBUG_PRINT enabled for code review. The standard practice is to enable it locally, and disable it when sending out for code review, so you don't forget to remove it when landing the code. Personally, I often have a local DO_NOT_SUBMIT change that I keep rebasing.

Reply:
Removed this and using GRAPH_DUMP